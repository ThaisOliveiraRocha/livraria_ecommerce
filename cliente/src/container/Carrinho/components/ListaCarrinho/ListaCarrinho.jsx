import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import { TituloTopo, Item, ImageTrash } from "./ListaCarrinho.styles";

import { connect } from "react-redux";
import Text from "../../../../components/Text";
import { showModal } from "../../../../store/actions";
import { withRouter } from "react-router-dom";

const ListaCarrinho = ({ item, showModal }) => (
  <TituloTopo>
    <Item>
      <Text>{item.titulo}</Text>
    </Item>
    <Item>
      <Text>{item.qtd}</Text>
    </Item>
    <Item>
      <Text>R$ {item.preco}</Text>
    </Item>
    <Item>
      <ImageTrash
        src={require("../../../../assets/images/excluir.png")}
        onClick={() => showModal(item.titulo)}
      />
    </Item>
  </TituloTopo>
);

ListaCarrinho.propTypes = {
  // bla: PropTypes.string,
};

const mapDispatchToProps = dispatch => ({
  showModal: titulo => {
    dispatch(showModal(titulo, true));
  }
});

export default withRouter(connect(null, mapDispatchToProps)(ListaCarrinho));
