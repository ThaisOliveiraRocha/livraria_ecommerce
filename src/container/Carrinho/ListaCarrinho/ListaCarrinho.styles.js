import styled from "styled-components";

export const BodyCart = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
`;

export const TableCart = styled.table`
  width: 60%;
  border-collapse: collapse;
  margin-top: 100px;
`;

export const TituloTopo = styled.tr`
    
`;

export const Cabecalho = styled.th`
  height: 50px;
  border: 1px solid #dddddd;
`;

export const Item = styled.td`
  height: 50px;
  text-align: center;
  border: 1px solid #dddddd;
`;

export const ButtonContainer = styled.div`
    width: 60%;
    display: flex;
    align-items: center;    
    justify-content: flex-end;
    font-size: 16px;
    font-weight: 600;
    color: #ffffff;
`;

export const ButtonFinalizar = styled.div`
    width: 100px;
    height: 30px;
    margin: 10px;

    background-color: #00BA08;

    display: flex;
    align-items: center;
    justify-content: center;
`;

export const ButtonCancelar = styled.div`
    width: 100px;
    height: 30px;
    margin: 10px 0 10px 10px;

    background-color: #CA1900;

    display: flex;
    align-items: center;
    justify-content: center;
`;
