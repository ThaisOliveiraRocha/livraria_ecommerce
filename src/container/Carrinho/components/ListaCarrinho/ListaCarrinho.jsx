import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import { 
  TableCart, 
  TituloTopo, 
  Cabecalho, 
  Item, 
  BodyCart, 
  ButtonContainer, 
  ButtonCancelar, 
  ButtonFinalizar,
  Texto 
} from "./ListaCarrinho.styles";

class ListaCarrinho extends PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      hasError: false
    };
  }

  render() {
    if (this.state.hasError) {
      return <h1>Something went wrong.</h1>;
    }
    return (
      <BodyCart>
        <TableCart>
          <TituloTopo>
            <Cabecalho>Título</Cabecalho>
            <Cabecalho>Preço</Cabecalho>
            <Cabecalho>Quantidade</Cabecalho>
            <Cabecalho>Descartar</Cabecalho>
          </TituloTopo>
          <TituloTopo>
            <Item>Nome</Item>
            <Item>preco</Item>
            <Item>qtd</Item>
            <Item>btn</Item>
          </TituloTopo>
          <TituloTopo>
            <Item>Nome</Item>
            <Item>preco</Item>
            <Item>qtd</Item>
            <Item>btn</Item>
          </TituloTopo>
        </TableCart>

        <ButtonContainer>
          <Texto>Total da Compra: </Texto>
          <ButtonFinalizar type="button">Finalizar</ButtonFinalizar>
          <ButtonCancelar type="button">Cancelar</ButtonCancelar>
        </ButtonContainer>
      </BodyCart>
    );
  }
}

ListaCarrinho.propTypes = {
  // bla: PropTypes.string,
};

ListaCarrinho.defaultProps = {
  // bla: 'test',
};

export default ListaCarrinho;
