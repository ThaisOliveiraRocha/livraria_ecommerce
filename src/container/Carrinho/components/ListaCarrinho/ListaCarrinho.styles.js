import styled from "styled-components";

export const BodyCart = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
`;

export const TableCart = styled.table`
  width: 60%;
  border-collapse: collapse;
  margin-top: 100px;
`;

export const TituloTopo = styled.tr`
    
`;

export const Cabecalho = styled.th`
  height: 50px;
  border: 1px solid #dddddd;
`;

export const Item = styled.td`
  height: 50px;
  text-align: center;
  border: 1px solid #dddddd;
`;

export const Input = styled.input`
  font-size: 16px;
  width: 50px;
`;

export const ButtonContainer = styled.div`
    width: 60%;
    display: flex;
    align-items: center;    
    justify-content: flex-end;
`;

export const Texto = styled.span`
  font-size: 16px;
  font-weight: 600;
  color: black;
`;

export const TotalContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: flex-end;
  width: 60%;
  margin: 10px 0;
`;

export const ImageTrash = styled.img`
  width: 20px;
  height: auto;
  cursor: pointer;

  &:active {
    transform: scale(0.9, 0.9);
  }
`;