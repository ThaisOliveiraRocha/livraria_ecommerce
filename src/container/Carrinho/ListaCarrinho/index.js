import ListaCarrinho from './ListaCarrinho';

export default ListaCarrinho;
