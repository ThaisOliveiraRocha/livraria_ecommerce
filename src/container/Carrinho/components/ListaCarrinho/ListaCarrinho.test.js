import React from "react";
import { shallow } from "enzyme";
import ListaCarrinho from "./ListaCarrinho";

describe("<ListaCarrinho />", () => {
  it("should render without crashing", () => {
    shallow(<ListaCarrinho />);
  });
  it("should render basic infos", () => {
    const wrapper = shallow(<ListaCarrinho />);
    expect(wrapper).toMatchSnapshot();
  });
  
});
