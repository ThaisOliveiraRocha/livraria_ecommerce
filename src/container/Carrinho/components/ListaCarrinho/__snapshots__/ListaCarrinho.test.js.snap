// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListaCarrinho /> should render basic infos 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
